src/appsettings/validators.py

```diff
@@ -67,3 +69,43 @@ def __call__(self, value):
             if not isinstance(element, self.value_type):
                 params = {"key": key, "value": element, "type": self.value_type.__name__}
                 raise ValidationError(self.message, params=params)
+
+
+class FileValidator(object):
+    """Validator which checks file existence and open mode permissions."""
+
+    not_exist_message = "File %(value)s does not exist."
+    no_perms_message = "You have no permission to open %(value)s with mode %(mode)s."
+    unexpected_error_message = "Unexpected error occurred while opening the file %(value)s: %(exception)s"
+
+    def __init__(
+        self, exists=True, mode="r", not_exist_message=None, no_perms_message=None, unexpected_error_message=None
+    ):
+        """
+        Try to open a file using a given ``mode``.
+
+        Args:
+            exists (bool): Require existence of the file
+            mode (str): ``mode`` argument for ``open`` builtin function.
+            not_exist_message (str): Error message (containing ``value``)
+            no_perms_message (str): Error message (containing ``value`` and ``mode``)
+            unexpected_error_message (str): Error message (containing ``value`` and ``exception``)
+        """
+        self.exists = exists
+        self.mode = mode
+        self.not_exist_message = not_exist_message or self.not_exist_message
+        self.no_perms_message = no_perms_message or self.no_perms_message
+        self.unexpected_error_message = unexpected_error_message or self.unexpected_error_message
```

Comment:
When there are multiple messages, I'd rather have them joined in a dictionary.

Reply:
You mean to pack all the variables into one dict?
```
class FileValidator(object):

    error_messages = dict(
        not_exist_message = "...",
        no_perms_message = "...",
        unexpected_error_message = "...",
    )

    def __init__(..., error_messages=None):
        self.error_messages = error_messages or self.error_messages
```
But then you are not able to modify just one error message.

Reply:
> You mean to pack all the variables into one dict?

Yes.

> But then you are not able to modify just one error message.

You can merge the dictionaries.

Reply:
> You can merge the dictionaries.

I don't like it. In your case there will have to be some dictionary keys in the documentation and all becomes less clean. Should I throw an exception on invalid key? Should I ignore it?

Reply:
Ignore. It's a common approach in Django for error messages, so it's not something entirely unexpected.

Reply:
Done.